ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/RetryableException.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.engine.algorithms.remote;
+
+import java.io.IOException;
+
+import org.opensearch.OpenSearchStatusException;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.rest.RestStatus;
+
+public class RetryableException extends OpenSearchStatusException {
```

Comment:
Is it possible that user can see `retry...` anywhere in the response if return this exception?

Reply:
If the retry succeed, then users won't see it in the response. If the request fails after timeout or exceed max retry times, users can see all exceptions during the retries.
By setting the logger to debug level, users can see the retry log and exception information. (logged at should retry)

Reply:
I mean when OpenSearchStatusException is returned to user, the response is similar to below:
```
{
  "error": {
    "root_cause": [
      {
        "type": "status_exception",
        "reason": "Model is disabled."
      }
    ],
    "type": "status_exception",
    "reason": "error message"
  },
  "status": 403
}
```
What if the retryableException is returned to user? Will it show `retry_exception` or still `status_exception`?

Reply:
Here is a real response body fails with max_retry_times=2
Do you think we need to change it?
```
[{'index': {'_index': 'scifact',
   '_id': '967',
   'status': 400,
   'error': {'type': 'retryable_exception',
    'reason': 'retryable_exception: Error from remote service: The request was denied due to remote server throttling.',
    'suppressed': [{'type': 'retryable_exception',
      'reason': 'retryable_exception: Error from remote service: The request was denied due to remote server throttling.'},
     {'type': 'retryable_exception',
      'reason': 'retryable_exception: Error from remote service: The request was denied due to remote server throttling.'}]}}}]
```

Reply:
Changed it to SageMakerThrottlingException, now we'll get this:
```
[{'index': {'_index': 'scifact',
   '_id': '897',
   'status': 400,
   'error': {'type': 'sage_maker_throttling_exception',
    'reason': 'sage_maker_throttling_exception: Error from remote service: The request was denied due to remote server throttling.',
    'suppressed': [{'type': 'sage_maker_throttling_exception',
      'reason': 'sage_maker_throttling_exception: Error from remote service: The request was denied due to remote server throttling.'},
     {'type': 'sage_maker_throttling_exception',
      'reason': 'sage_maker_throttling_exception: Error from remote service: The request was denied due to remote server throttling.'}]}}}]
```